### DISCUSSION TITLE
DynamicsWebApi v2

### DISCUSSION THREAD

**Discussion by AleksandrRogov** [AleksandrRogov/DynamicsWebApi#137] [Announcements]
I started actively working on v2 of DynamicsWebApi. Though, due to limited free time that I have, I can only do it for some time on weekends, so it may take awhile until the beta version is released. 

### Need help

If you'd like to volunteer to help me out, I am happy to accept your help! Just let me know beforehand in this Discussion or in any issue tagged with `v2`. 

**I desperately need help with documentation and examples.**

### Development Progress

You can check the progress of development by opening [DynamicsWebApi v2 Project](https://github.com/users/AleksandrRogov/projects/2) on GitHub. Also, you can prepare yourself for breaking changes listed in #135 . In addition to that, I will be releasing patches to v1 which will include deprecation notices for some parameters and their corresponding replacements if they are possible to do in v1.

### Suggestions

If you have any suggestions on what to include in v2 - now is the best time to let me know! Let me know if you dislike something in v1 and what could be done with it, or even what you like in it, so I don't "accidently" remove it!

### Thank you

I am extremely excited about the v2 because it brings so many improvements and cool features that I can't wait to share with everyone.

Best Regards,
Aleksandr

**Comment by AidanLaycock** [comment]
Hi, I am just looking at picking this package up for a project I'm about to start, is there a rough ETA on V2 being released? As it would be great to start using that rather than V1 if it introduces breaking changes.

**Reply by AleksandrRogov** [reply]
hi!
I was planning to make a beta release in the beginning of June, so everyone (including myself) could test it out before a full release.
I do have an alpha package ready that I could publish on NPM. It does not include #145 . It works well, has a high test coverage and [documentation](../tree/typescript#readme) is pretty much done. There still may be breaking changes though, but I don't think they will be as significant as between v1 and v2.
Let me know!
